spring-integration-core/src/main/java/org/springframework/integration/handler/advice/RequestHandlerCircuitBreakerAdvice.java

```diff
@@ -95,9 +95,10 @@ private AtomicInteger getFailures() {
 
 		private static final long serialVersionUID = 1L;
 
-		public CircuitBreakerOpenException(String message) {
+		private CircuitBreakerOpenException(String message) {
```

Comment:
Technically this exception should be moved on top.
It might be useful for end-application rely on the exception thrown.
Otherwise, please, explain the reason to hide it here and throw to the caller from `doInvoke()`.